src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -99,3 +99,67 @@ SHARED_API int hostfxr_main(const int argc, const pal::char_t* argv[])
     return muxer.execute(argc, argv);
 }
 
+//
+// Determines the directory location of the SDK accounting for global.json and multi-level lookup policy.
+//
+// Invoked via MSBuild SDK resolver to locate SDK props and targets from an msbuild other than the one
+// bundled by the CLI.
+//
+// Parameters:
+//    global_json_start_dir - The directory where the search for global.json 
+//    buffer_size           - The size of the buffer argument in pal::char_t units (see encoding note below)
+//    buffer               -  The buffer where the located SDK path will be written.
+//
+// Return value:
+//   <0 - Invalid argument
+//   0  - SDK could not be found.
+//   >0 - The number of characters (including null terminator) required to store the located SDK.
+//
+//   If resolution succeeds and the positive return value is less than or equal to buffer_size (i.e. the
+//   the buffer is large enough), then the resolved SDK path is copied to the buffer and null terminated.
+//   Otherwise, no data is written to the buffer.
+//
+// String encoding:
+//   Windows     - UTF-16 (pal::char_t is 2 byte wchar_t)
```

Comment:
The immediate use case is for Windows for VS, but I think VS for Mac will want to do likewise. Any other environment using its own msbuild but needing to pick up the tasks and targets from the CLI would have the same requirement.